pkg/watermark/wmb/wmb.go

```diff
@@ -58,3 +58,55 @@ func DecodeToWMB(b []byte) (WMB, error) {
 	}
 	return v, nil
 }
+
+type WMBChecker struct {
+	counter int
+	max     int
+	w       WMB
+}
+
+// NewWMBChecker returns a WMBChecker to check if the wmb is idle.
+// If all the iterations get the same wmb, the wmb is
+func NewWMBChecker(numOfIteration int) WMBChecker {
```

Comment:
the wmb is?

Reply:
WMB is short for "watermark barrier" ... which is basically a watermark with idle=true/false information. Open for better name. :-)

Reply:
No, I meant in line 69 the comment is not complete

Reply:
updated :-D